packages/matter.js/src/protocol/ControllerCommissioningHandler.ts

```diff
@@ -0,0 +1,783 @@
+                    newRegulatoryConfig: locationCapability,
+                    countryCode
+                }));
+            } else {
+                this.ensureGeneralCommissioningSuccess("setRegulatoryConfig", regulatoryResult);
+            }
+            return {
+                code: CommissioningStepResultCode.Success,
+                breadcrumb: this.lastBreadcrumb,
+            }
+        }
+        return {
+            code: CommissioningStepResultCode.Skipped,
+            breadcrumb: this.lastBreadcrumb,
+        }
+    }
+
+    /**
+     * Step 5 - 2
+     * Commissioner SHOULD configure UTC time, timezone, and DST offset, if the Commissionee supports the time cluster.
+     * The order of configuration of this information is not critical. The UTC time is configured using SetUtcTime
+     * command (see Section 11.16.9.1, “SetUtcTime Command”) while timezone and DST offset are set through TimeZone
+     * (see Section 11.16.8.6, “TimeZone Attribute”) and DstOffset attribute (see Section 11.16.8.7,
+     * “DSTOffset Attribute”), respectively.
+     */
+    private async synchronizeTime() {
+        if (this.collectedCommissioningData.rootServerList !== undefined && this.collectedCommissioningData.rootServerList.find(clusterId => clusterId.id === TimeSyncCluster.id)) {
+            logger.debug("Commissioning step 5.2: TimeSync cluster is supported");
+            // TODO: implement
+        }
+        return {
+            code: CommissioningStepResultCode.Skipped,
+            breadcrumb: this.lastBreadcrumb,
+        };
+    }
+
+    /**
+     * Step 6
+     * Commissioner SHALL establish the authenticity of the Commissionee as a certified Matter device (see Section
+     * 6.2.3, “Device Attestation Procedure”).
+     */
+    private async deviceAttestation() {
+        const operationalCredentialsClusterClient = this.getClusterClient(OperationalCredentials.Cluster);
+        const { certificate: deviceAttestation } = await operationalCredentialsClusterClient.certificateChainRequest({ certificateType: OperationalCredentials.CertificateChainType.DacCertificate });
+        // TODO: extract device public key from deviceAttestation
+        const { certificate: productAttestation } = await operationalCredentialsClusterClient.certificateChainRequest({ certificateType: OperationalCredentials.CertificateChainType.PaiCertificate });
+        // TODO: validate deviceAttestation and productAttestation
+        const { attestationElements, attestationSignature } = await operationalCredentialsClusterClient.attestationRequest({ attestationNonce: Crypto.getRandomData(32) });
+        // TODO: validate attestationSignature using device public key
+        if (deviceAttestation.length === 0 || productAttestation.length === 0 || attestationElements.length === 0 || attestationSignature.length === 0) {
+            // TODO: validate the data really
+            throw new Error("Invalid response from device");
```

Comment:
More specificity would be nice, even if just "see logs for details", but maybe that's covered by the TODO?

Reply:
Yes but I will still enhance that